src/test/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisTest.java

```diff
@@ -3421,4 +3422,113 @@ private void testWithFictitiousLoad2(LoadFlowParameters.BalanceType balanceType)
+                .setP0(1)
+                .setQ0(0)
+                .add();
+
+        SecurityAnalysisParameters securityAnalysisParameters = new SecurityAnalysisParameters();
+        securityAnalysisParameters.getLoadFlowParameters().setConnectedComponentMode(LoadFlowParameters.ConnectedComponentMode.ALL);
+
+        List<Contingency> contingencies = List.of(new Contingency("l13", new BranchContingency("l13")),
+                new Contingency("dummyLoad", new LoadContingency("dummyLoad")));
+
+        // Monitor branch in both components
+        List<StateMonitor> monitors = List.of(
+                new StateMonitor(ContingencyContext.all(), Set.of("l14", "l12", "l23", "lc12", "lc12Bis"), emptySet(), emptySet())
+        );
+
+        SecurityAnalysisResult results = runSecurityAnalysis(network, contingencies, monitors, securityAnalysisParameters);
+        NetworkResult preContingencyResults = results.getPreContingencyResult().getNetworkResult();
+        NetworkResult postContingencyResultsl13 = results.getPostContingencyResults().get(0).getNetworkResult();
+        NetworkResult postContingencyResultslc12Bis = results.getPostContingencyResults().get(1).getNetworkResult();
+
+        // Result for base case is available for all components
+        assertEquals(0.084, preContingencyResults.getBranchResult("l14").getP1(), LoadFlowAssert.DELTA_POWER);
+        assertEquals(1.417, preContingencyResults.getBranchResult("l12").getP1(), LoadFlowAssert.DELTA_POWER);
+        assertEquals(0.417, preContingencyResults.getBranchResult("l23").getP1(), LoadFlowAssert.DELTA_POWER);
+        assertEquals(0.493, preContingencyResults.getBranchResult("lc12").getP1(), LoadFlowAssert.DELTA_POWER);
+        assertEquals(0.493, preContingencyResults.getBranchResult("lc12Bis").getP1(), LoadFlowAssert.DELTA_POWER);
+
+        // Result for post contingency on l13 is available for part of the network on which the contingency has an impact
+        assertEquals(1.021, postContingencyResultsl13.getBranchResult("l14").getP1(), LoadFlowAssert.DELTA_POWER);
+        assertEquals(2.311, postContingencyResultsl13.getBranchResult("l12").getP1(), LoadFlowAssert.DELTA_POWER);
+        assertEquals(1.310, postContingencyResultsl13.getBranchResult("l23").getP1(), LoadFlowAssert.DELTA_POWER);
+
+        // No impact on this component, no results
+        assertNull(postContingencyResultsl13.getBranchResult("lc12"));
+        assertNull(postContingencyResultsl13.getBranchResult("lc12Bis"));
+
+        // Result for post contingency on dummyLoad is available for part of the network on which the contingency has an impact
+        assertEquals(0.498, postContingencyResultslc12Bis.getBranchResult("lc12").getP1(), LoadFlowAssert.DELTA_POWER);
+        assertEquals(0.498, postContingencyResultslc12Bis.getBranchResult("lc12Bis").getP1(), LoadFlowAssert.DELTA_POWER);
+    }
+
+    @Test
+    void multiComponentSaTestContingencyBothComponentsAndOperatorStrategy() {
+
```

Comment:
What happens if a remedial action reconnects two networks that are initially isolated ?

Is this supported ? Should this be a listed as a limitation of this feature ?

Reply:
I think behavior might depend on the network being isolated only after contingency and or also in the base case. This might be a good idea to test this case specifically yes.